### DISCUSSION TITLE
All pages become dynamic when built

### DISCUSSION THREAD

**Discussion by killjoy2013** [vercel/next.js#68405] [Help]
### Summary

I have layout.tsx

```ts
 <body>
        <AppRouterCacheProvider>
          <NotistackProvider maxSnack={3}>
            <SessionProvider>
              <Stack>
                <Providers>{children}</Providers>
              </Stack>
            </SessionProvider>
          </NotistackProvider>
        </AppRouterCacheProvider>
      </body>
```

when I build the app, static & dynamic pages created as expected;

<img width="506" alt="Screenshot 2024-08-01 at 17 56 28" src="https://github.com/user-attachments/assets/e611d7cd-9983-4684-b98c-635ed19d8410">

However, I need to add a `<Header/>` server component with sign-in & sing-out logic. Layout.tsx become

```ts
<body>
        <AppRouterCacheProvider>
          <NotistackProvider maxSnack={3}>
            <SessionProvider>
              <Stack>
                <Header />
                <Providers>{children}</Providers>
              </Stack>
            </SessionProvider>
          </NotistackProvider>
        </AppRouterCacheProvider>
      </body>
```

Altough the application runs OK, all the pages become dynamic unnecessarily;

<img width="489" alt="Screenshot 2024-08-01 at 18 04 04" src="https://github.com/user-attachments/assets/c0e814af-73c6-4188-9c8e-c1772b93089d">

How to keep the `<Header/>` and still separate static & dynamic pages correctly?

Thanks

**Comment by icyJoseph** [comment]
Either... https://nextjs.org/docs/app/api-reference/next-config-js/partial-prerendering which I would guess could work

Or, refactor how the header component works, not entirely sure, but you probably use headers or cookies within it?
